File: src/main/scala/eightyDays/scala211/Benchmarks.scala
package eightyDays.scala211

import eightyDays.scala211.bank.Bank
import eightyDays.Testdata
import org.openjdk.jmh.annotations.{Benchmark, Scope, Setup, State}

object Benchmarks {
  val testdata = new Testdata
  val bigBank = testdata.personsS(20000).foldLeft(Bank("Simple bank")) {
    (bank, person) =>
      bank.addPartner(person)._2
  }
}

@State(Scope.Benchmark)
class Benchmarks {

  @Setup
  def prepare(): Unit = Benchmarks.bigBank

  @Benchmark def measureSearch = {
    Benchmarks.bigBank.find(_.name == "Muster")
  }
}

File: src/main/java/eightyDays/java8/Benchmarks.java
package eightyDays.java8;

import eightyDays.java8.bank.Bank;
import eightyDays.java8.bank.partner.Partner;
import eightyDays.scala211.Testdata;
import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.Scope;
import org.openjdk.jmh.annotations.State;

public class Benchmarks {

    @State(Scope.Benchmark)
    public static class Provider {
        Testdata testdata = new Testdata();

        Bank bigBank = bigBank();

        private Bank bigBank() {
            Bank bigBank = new Bank("Simple bank");
            testdata.personsJ(20_000).forEach(bigBank::add);
            return bigBank;
        }
    }

    @Benchmark
    public void measureSearch(Provider provider) {
        Bank out = provider.bigBank;
        out.find(Partner.byName("Muster"));
    }
}
